File: ios/drycleanposapp/BrotherPrinter.m
// BrotherPrinter.m
#import <React/RCTBridgeModule.h>

@interface RCT_EXTERN_MODULE(BrotherPrinter, NSObject)

RCT_EXTERN_METHOD(searchPrinters:
                  (RCTPromiseResolveBlock)resolve
                  rejecter:(RCTPromiseRejectBlock)reject)

RCT_EXTERN_METHOD(printLabel:(NSString *)printerIP
                  text:(NSString *)text
                  width:(nonnumber *)width
                  height:(nonnumber *)height
                  resolver:(RCTPromiseResolveBlock)resolve
                  rejecter:(RCTPromiseRejectBlock)reject)

@end

File: src/modules/BrotherPrinter.ts
// Create this file to interface with the native module
import { requireNativeModule } from 'expo-modules-core';

// This loads the native module object from the JSI framework
const BrotherPrinter = requireNativeModule('BrotherPrinter');

export default {
  searchPrinters: () => BrotherPrinter.searchPrinters(),
  printLabel: (printerIP: string, text: string, width: number, height: number) => 
    BrotherPrinter.printLabel(printerIP, text, width, height)
};